### PR TITLE
Set retention policy while dlq topic creation

### PR DESCRIPTION
**Pull request by sakshi-goyal-razorpay**
# Description
Set up retention policy for newly created dlq topics

Fixes https://razorpay.atlassian.net/browse/METRO-137

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Tested on local by creating a dlq topic, and checking its config. Also messages were purged once garbage collection happened.

# Is it a breaking change?
No

## Screenshots:
Screenshot 2022-09-13 at 1 20 31 PM

# Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added unit tests that prove my fix is effective or that my feature works (if applicable)
- [x] I have added integration tests for the new feature (if applicable)
- [x] I have manually tested my code to the best of my abilities.

### CODE REVIEW

File: tests/integration/messagebroker/messagebroker_kafka_test.go [hunk 1]
```diff
@@ -314,6 +314,7 @@ func Test_ResetAutoOffsetForConsumer(t *testing.T) {
 	aresp, err := admin.CreateTopic(context.Background(), messagebroker.CreateTopicRequest{
 		Name:          topic,
 		NumPartitions: 1,
+		Config:        map[string]string{"retention.ms": "10000"},
```
Comment (sumitkrzp) [inline tests/integration/messagebroker/messagebroker_kafka_test.go:317]: @sakshi-goyal-razorpay why "retention.bytes" is not added in the config map

Comment (sakshi-goyal-razorpay) [reply tests/integration/messagebroker/messagebroker_kafka_test.go:317]: It is just a test case, so added just a single config. But sure, will add other too

File: internal/topic/core_test.go [hunk 2]
```diff
@@ -29,14 +29,28 @@ func TestCore_CreateTopic(t *testing.T) {
...
-	mockBrokerStore.EXPECT().GetAdmin(gomock.Any(), messagebroker.AdminClientOptions{}).Return(mockAdmin, nil)
-	mockAdmin.EXPECT().CreateTopic(gomock.Any(), messagebroker.CreateTopicRequest{dTopic.Name, DefaultNumPartitions}).Return(messagebroker.CreateTopicResponse{}, nil)
-	mockTopicRepo.EXPECT().Exists(gomock.Any(), common.GetBasePrefix()+"topics/test-project/test-topic")
-	mockTopicRepo.EXPECT().Save(gomock.Any(), dTopic)
-	err := topicCore.CreateTopic(ctx, dTopic)
-	assert.Nil(t, err)
+
+	tests := []struct {
+		topicModel *Model
+	}{
```
Comment (vnktram) [inline internal/topic/core_test.go:35]: Please fix formatting using a linter

Comment (sakshi-goyal-razorpay) [reply internal/topic/core_test.go:35]: linter only made this change. This is how we have defined struct formatting in the past as well.
https://github.com/razorpay/metro/blob/master/internal/subscriber/subscriberimpl_test.go#L48

**Review by vnktram — APPROVED** [review]

**Review by akashgoyal96 — APPROVED** [review]
LGTM

**Review by sumitkrzp — APPROVED** [review]
LGTM

**Pull request merged**